plugins/modules/v_switch.py

```diff
@@ -0,0 +1,486 @@
+
+    return result
+
+
+def is_all_servers_ready(result, dummy):
+    return all(server['status'] == 'ready' for server in result['server'])
+
+
+def add_servers(module, id_, servers):
+    url = '{0}/{1}/server'.format(V_SWITCH_BASE_URL, id_)
+    headers = {'Content-type': 'application/x-www-form-urlencoded'}
+    data = get_x_www_form_urlenconded_dict_from_list('server', servers)
+    result, error = fetch_url_json(
+        module,
+        url,
+        data=urlencode(data),
+        headers=headers,
+        method='POST',
+        # TODO: missing INVALID_INPUT, NOT_FOUND, VSWITCH_NOT_AVAILABLE, VSWITCH_PER_SERVER_LIMIT_REACHED
+        accept_errors=[
+            'SERVER_NOT_FOUND',
+            'VSWITCH_VLAN_NOT_UNIQUE',
+            'VSWITCH_IN_PROCESS',
+            'VSWITCH_SERVER_LIMIT_REACHED',
+        ],
+        allow_empty_result=True,
+        allowed_empty_result_status_codes=(201,),
+    )
+    if error == 'SERVER_NOT_FOUND':
+        # information about which servers are failing is only there
+        module.fail_json(msg=result['error']['message'])
+    elif error == 'VSWITCH_VLAN_NOT_UNIQUE':
+        # information about which servers are failing is only there
+        module.fail_json(msg=result['error']['message'])
+    elif error == 'VSWITCH_IN_PROCESS':
+        module.fail_json(msg='There is a update running, therefore the vswitch can not be updated')
+    elif error == 'VSWITCH_SERVER_LIMIT_REACHED':
+        module.fail_json(msg='The maximum number of servers is reached for this vSwitch')
+
+    # TODO: add and delete with `wait=false`
+    wait_condition = is_all_servers_ready if module.params['wait'] else None
+    return get_v_switch(module, id_, wait_condition)
+
+
+def delete_servers(module, id_, servers):
+    url = '{0}/{1}/server'.format(V_SWITCH_BASE_URL, id_)
+    headers = {'Content-type': 'application/x-www-form-urlencoded'}
+    data = get_x_www_form_urlenconded_dict_from_list('server', servers)
+    result, error = fetch_url_json(
+        module,
+        url,
+        data=urlencode(data),
+        headers=headers,
+        method='DELETE',
+        # TODO: missing INVALID_INPUT, NOT_FOUND
+        accept_errors=['SERVER_NOT_FOUND', 'VSWITCH_IN_PROCESS'],
+        allow_empty_result=True,
+    )
+    if error == 'SERVER_NOT_FOUND':
+        # information about which servers are failing is only there
+        module.fail_json(msg=result['error']['message'])
+    elif error == 'VSWITCH_IN_PROCESS':
+        module.fail_json(msg='There is a update running, therefore the vswitch can not be updated')
+
+    wait_condition = is_all_servers_ready if module.params['wait'] else None
+    return get_v_switch(module, id_, wait_condition)
+
+
+def get_servers_to_delete(current_servers, desired_servers):
+    return [
+        server['server_ip']
+        for server in current_servers
+        if server['server_ip'] not in desired_servers
+        and str(server['server_number']) not in desired_servers
```

Comment:
What about the server's main IPv6?

Reply:
Ouch, nice catch! Solved in 64da0ee545e8c9d0cc6a1fc04c9116d74866edd9